libcasr/src/js.rs

```diff
@@ -0,0 +1,379 @@
+        if sttr.is_err() {
+            panic!("{}", sttr.err().unwrap());
+        }
+        let mut stacktrace = sttr.unwrap();
+
+        init_ignored_frames!("js");
+        stacktrace.filter();
+
+        assert_eq!(
+            stacktrace[0].debug.file,
+            "/fuzz/node_modules/jpeg-js/lib/decoder.js".to_string()
+        );
+        assert_eq!(stacktrace[0].debug.line, 1110);
+        assert_eq!(stacktrace[0].debug.column, 13);
+        assert_eq!(stacktrace[0].function, "Object.decode".to_string());
+        assert_eq!(stacktrace[1].debug.file, "/fuzz/FuzzTarget.js".to_string());
+        assert_eq!(stacktrace[1].debug.line, 6);
+        assert_eq!(stacktrace[1].debug.column, 14);
+        assert_eq!(stacktrace[1].function, "fuzz".to_string());
+        assert_eq!(
+            stacktrace[2].debug.file,
+            "/fuzz/node_modules/@jazzer.js/core/core.ts".to_string()
+        );
+        assert_eq!(stacktrace[2].debug.line, 335);
+        assert_eq!(stacktrace[2].debug.column, 15);
+        assert_eq!(stacktrace[2].function, "result".to_string());
+        assert_eq!(
+            stacktrace[3].debug.file,
+            "/home/user/test_js_stacktrace/main.js".to_string()
+        );
+        assert_eq!(stacktrace[3].debug.line, 1);
+        assert_eq!(stacktrace[3].debug.column, 2017);
+        assert_eq!(stacktrace[3].function, "Worker.fuzz [as fn]".to_string());
+        assert_eq!(
+            stacktrace[4].debug.file,
+            "/home/user/.nvm/versions/node/v16.15.1/lib/node_modules/jsfuzz/build/src/worker.js"
+                .to_string()
+        );
+        assert_eq!(stacktrace[4].debug.line, 55);
+        assert_eq!(stacktrace[4].debug.column, 30);
+        assert_eq!(stacktrace[4].function, "process.<anonymous>".to_string());
+        assert_eq!(
+            stacktrace[5].debug.file,
+            "/home/user/node/offset.js".to_string()
+        );
+        assert_eq!(stacktrace[5].debug.line, 3);
+        assert_eq!(stacktrace[5].debug.column, 37);
+        assert_eq!(stacktrace[5].function, "".to_string());
+        assert_eq!(
+            stacktrace[6].debug.file,
+            "/home/user/node/offset.js".to_string()
+        );
+        assert_eq!(stacktrace[6].debug.line, 3);
+        assert_eq!(stacktrace[6].debug.column, 7);
+        assert_eq!(stacktrace[6].function, "eval".to_string());
+        assert_eq!(stacktrace[7].debug.file, "/fuzz/FuzzTarget.js".to_string());
+        assert_eq!(stacktrace[7].debug.line, 12);
+        assert_eq!(stacktrace[7].debug.column, 13);
+        assert_eq!(stacktrace[7].function, "eval at g".to_string());
+        assert_eq!(
+            stacktrace[8].debug.file,
+            "/.svelte-kit/runtime/components/layout.svelte".to_string()
+        );
+        assert_eq!(stacktrace[8].debug.line, 8);
+        assert_eq!(stacktrace[8].debug.column, 41);
+        assert_eq!(stacktrace[8].function, "eval".to_string());
+    }
+
+    #[test]
+    fn test_js_exception() {
```

Comment:
I think we should to merge both tests and extract exception from full error message

Reply:
Done